### PR TITLE
fix bug in fallback operations for probabilisticnetworkcomposites

### LINKED ISSUE #864: Error when calling fallback operations for probabilistic networkcomposites
A bug slipped through the new breaking release of MLJBase. As a result of this bug, calling the fallback methods of operations such as  `predict_mode`, `predict_mean`, and `predict_median` on a `ProbabilisticNetworkcomposite` or `ProbabilisticComposite` model yields a wrong result or throws a confusing error. The following example showcases this.

```julia
julia> using MLJBase, MLJModels, StableRNGs

julia> rng = StableRNG(212)
StableRNGs.LehmerRNG(state=0x000000000000000000000000000001a9)

julia> X, y = MLJBase.table(rand(rng, 4, 4)), rand(rng, 4)
(Tables.MatrixTable{Matrix{Float64}} with 4 rows, 4 columns, and schema:
 :x1  Float64
 :x2  Float64
 :x3  Float64
 :x4  Float64, [0.8114256800965156, 0.2168426604870377, 0.23161321549413416, 0.26552646144829795])

julia> network_composite_model = FeatureSelector() |> ConstantRegressor()
ProbabilisticPipeline(
  feature_selector = FeatureSelector(
        features = Symbol[],
        ignore = false),
  constant_regressor = ConstantRegressor(
        distribution_type = Distributions.Normal),
  cache = true)

julia> mach = fit!(machine(network_composite_model, X, y), verbosity=0)
trained Machine; does not cache data
  model: ProbabilisticPipeline(feature_selector = FeatureSelector(features = Symbol[], …), …)
  args:
    1:  Source @806 ⏎ Table{AbstractVector{Continuous}}
    2:  Source @648 ⏎ AbstractVector{Continuous}

julia> predict(mach, X) # this is fine
4-element Vector{Distributions.Normal{Float64}}:
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)

julia> predict_mode(mach, X) # this throws a confusing error
ERROR: ArgumentError: mode is not defined for empty collections

```

```
MLJBase v0.21.0
MLJModels v0.16.0
```
similar to https://github.com/alan-turing-institute/MLJ.jl/issues/987

### PR DESCRIPTION
**Pull request by OkonSamuel**
fixes #864, https://github.com/alan-turing-institute/MLJ.jl/issues/987

**Before PR**
```julia 
julia> using MLJBase, MLJModels, StableRNGs

julia> rng = StableRNG(212)
StableRNGs.LehmerRNG(state=0x000000000000000000000000000001a9)

julia> X, y = MLJBase.table(rand(rng, 4, 4)), rand(rng, 4)
(Tables.MatrixTable{Matrix{Float64}} with 4 rows, 4 columns, and schema:
 :x1  Float64
 :x2  Float64
 :x3  Float64
 :x4  Float64, [0.8114256800965156, 0.2168426604870377, 0.23161321549413416, 0.26552646144829795])

julia> network_composite_model = FeatureSelector() |> ConstantRegressor()
ProbabilisticPipeline(
  feature_selector = FeatureSelector(
        features = Symbol[],
        ignore = false),
  constant_regressor = ConstantRegressor(
        distribution_type = Distributions.Normal),
  cache = true)

julia> mach = fit!(machine(network_composite_model, X, y), verbosity=0)
trained Machine; does not cache data
  model: ProbabilisticPipeline(feature_selector = FeatureSelector(features = Symbol[], …), …)
  args:
    1:  Source @806 ⏎ Table{AbstractVector{Continuous}}
    2:  Source @648 ⏎ AbstractVector{Continuous}

julia> predict(mach, X) # this is fine
4-element Vector{Distributions.Normal{Float64}}:
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)

julia> predict_mode(mach, X) # this throws a confusing error
ERROR: ArgumentError: mode is not defined for empty collections
```

**After PR**
```julia
julia> using MLJBase, MLJModels, StableRNGs

julia> rng = StableRNG(212)
StableRNGs.LehmerRNG(state=0x000000000000000000000000000001a9)

julia> X, y = MLJBase.table(rand(rng, 4, 4)), rand(rng, 4)
(Tables.MatrixTable{Matrix{Float64}} with 4 rows, 4 columns, and schema:
 :x1  Float64
 :x2  Float64
 :x3  Float64
 :x4  Float64, [0.8114256800965156, 0.2168426604870377, 0.23161321549413416, 0.26552646144829795])

julia> network_composite_model = FeatureSelector() |> ConstantRegressor()
ProbabilisticPipeline(
  feature_selector = FeatureSelector(
        features = Symbol[],
        ignore = false),
  constant_regressor = ConstantRegressor(
        distribution_type = Distributions.Normal),
  cache = true)

julia> mach = fit!(machine(network_composite_model, X, y), verbosity=0)
trained Machine; does not cache data
  model: ProbabilisticPipeline(feature_selector = FeatureSelector(features = Symbol[], …), …)
  args:
    1:  Source @358 ⏎ Table{AbstractVector{Continuous}}
    2:  Source @667 ⏎ AbstractVector{Continuous}


julia> predict(mach, X) # this is fine
4-element Vector{Distributions.Normal{Float64}}:
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)
 Distributions.Normal{Float64}(μ=0.38135200438149636, σ=0.24892968488352138)

julia> predict_mode(mach, X) # this now works as expected.
4-element Vector{Float64}:
 0.38135200438149636
 0.38135200438149636
 0.38135200438149636
 0.38135200438149636
```

### CODE REVIEW

File: test/_models/simple_composite_model_2.jl [hunk 1]
```diff
@@ -0,0 +1,250 @@
+export SimpleDeterministicCompositeModel, SimpleDeterministicNetworkCompositeModel
+
```
Comment (ablaom) [inline test/_models/simple_composite_model_2.jl]: I'm confused. Aren't these already exported by test/_models/simple_composite_model.jl ?

File: test/operations.jl [hunk 2]
```diff
@@ -2,15 +2,84 @@ module TestOperations
...
+@testset "operations on network-composite models" begin
+    X = MLJBase.table(rand(4, 4))
+    y = rand(4)
+    m = fit!(machine(SimpleProbabilisticNetworkCompositeModel(), X, y), verbosity=0)
+    predictions = first(MLJBase.output_and_report(m.fitresult, :predict, X))
+    @test predict(m, X) == predictions
+    @test predict_mode(m, X) == mode.(predictions)
+    @test_throws ErrorException transform(m, X)
+end
+
```
Comment (ablaom) [inline test/operations.jl:69]: I think tests above can also be marked for removal. `*NetworkComposite` models are now deprecated.

**Review by ablaom — COMMENTED** [review]
Fine work. Great to have the extra testing. Just a little confused about the new `simple_composite_model_2.jl` as noted. Otherwise this looks good to go.

File: test/_models/simple_composite_model_2.jl (hunk 1 above)
Comment (OkonSamuel) [reply test/_models/simple_composite_model_2.jl]: That file is already deleted

**Comment by ablaom** [comment]
https://github.com/JuliaAI/MLJTestInterface.jl/issues/3

File: test/operations.jl (hunk 2 above)
Comment (OkonSamuel) [reply test/operations.jl:69]: I thought `NetworkComposite` models and `Composite` models are two different things. From my understanding `*NetworkComposite` is the new interface while `*Composite` and `*Surrogate` models are now deprecated and would be removed in the next breaking release.

Comment (ablaom) [reply test/operations.jl:69]: Oops. Brain cloud. Yes, it is the other way round! 😳

File: test/_models/simple_composite_model_2.jl (hunk 1 above)
Comment (ablaom) [reply test/_models/simple_composite_model_2.jl]: Got it.

**Review by ablaom — APPROVED** [review]

**Pull request merged**